### PR TITLE
plan: no need to pass the ctx to logicalOptRule

### PR DESCRIPTION
**Pull request by winoros**
Assign value may cause data race.
PTAL @hanfei1991 @zz-jason @lamxTyler

### CODE REVIEW

**Comment by zz-jason** [comment]
LGTM

**Comment by hanfei1991** [comment]
plan interface has `context()` function.

**Comment by coocood** [comment]
@winoros 
aggregationOptimizer has a context fields too.

File: plan/aggregation_push_down.go [hunk 1]
```diff
@@ -419,57 +419,57 @@ func (a *aggregationOptimizer) tryToEliminateAggregation(agg *LogicalAggregation
 func (a *aggregationOptimizer) convertAggToProj(agg *LogicalAggregation, ctx context.Context) *LogicalProjection {
```
Comment (hanfei1991) [inline plan/aggregation_push_down.go]: no need to pass `ctx`, which is contianed by agg

File: plan/aggregation_push_down.go [hunk 2]
```diff
@@ -398,7 +398,7 @@ func (a *aggregationOptimizer) aggPushDown(p LogicalPlan) LogicalPlan {
 // e.g. select min(b) from t group by a. If a is a unique key, then this sql is equal to `select b from t group by a`.
 // For count(expr), sum(expr), avg(expr), count(distinct expr, [expr...]) we may need to rewrite the expr. Details are shown below.
 // If we can eliminate agg successful, we return a projection. Else we return a nil pointer.
-func (a *aggregationOptimizer) tryToEliminateAggregation(agg *LogicalAggregation) *LogicalProjection {
+func (a *aggregationOptimizer) tryToEliminateAggregation(ctx context.Context, agg *LogicalAggregation) *LogicalProjection {
```
Comment (hanfei1991) [inline plan/aggregation_push_down.go]: no need to pass ctx

**Comment by coocood** [comment]
LGTM

**Comment by shenli** [comment]
Does this bug exist in release-1.0?

**Comment by winoros** [comment]
@shenli agg push down is disabled, maxMinElimination is not contained in release-1.0

**Pull request merged**